site/content/extend/plugins/webapp/best-practices.md

```diff
@@ -0,0 +1,48 @@
+- Examples: Jira project subscriptions to a channel; Zoom meeting posts to a channel
+
+2. Actions that apply to specific messages
+- Recommendation: Have your plugin register a [post dropdown menu component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPostDropdownMenuComponent) with some text, icon and an action function. This adds your action to the "More Actions" post menu dropdown.
+- Examples: Create or attach to Jira issue from a message; Copy a message to another channel; report an inappropriate message
+
+3. Actions related to files or images
+- Recommendation: Have your plugin register a [file upload method](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerFileUploadMethod) with some text, icon and an action function. This adds your action to the file upload menu.
+- Examples: File sharing from OneDrive or GDrive; Draw plugin for sketches
+
+4. Actions that apply to specific users
+- Recommendation: Have your plugin register a [popover user actions component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPopoverUserActionsComponent). This adds your action to the user profile popover. 
+- Examples: ..
+
+5. Attributes added for a user
+- Recommendation: Have your plugin register a [popover user attribute component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerPopoverUserAttributesComponent). This adds your attributes to the user profile popover. 
+- Examples: Custom User Attributes plugin
+
+6. Actions related to emoji and GIFs
+- Recommendation: Have your plugin add a component to the emoji picker. This is not yet supported, but some [work had previously started](https://github.com/mattermost/mattermost-server/issues/10412#issuecomment-481776595) with the issue currently opeend as Help Wanted
+- Examples: Bitmoji plugin; GIFs via Giphy or Gfycat
+
+7. Actions that apply to specific teams
+- Recommendation: Have your plugin register [left sidebar header component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerLeftSidebarHeaderComponent) with some text, icon and an action function. This adds your action above your team's channels in the sidebar.
+- Examples: Trello kanban board plugin, ..
+
+8. Quick links or status summaries of workflows ..??
+- Recommendation: Have your plugin register a [bottom team sidebar component](https://developers.mattermost.com/extend/plugins/webapp/reference/#registerBottomTeamSidebarComponent). This adds ... to ...
```

Comment:
Need a revision on when to best use a bottom team sidebar component

Reply:
IS an example how GithUb is at the bottom left?  Like this: 
![image](https://user-images.githubusercontent.com/915956/61008407-367fb500-a324-11e9-9b16-f127b3fd45bd.png)

Reply:
Yeah, GitHub plugin would be an example 👍